api/apiclient.go

```diff
@@ -498,11 +505,14 @@ func dialWebsocket(addr, path string, opts DialOpts, tlsConfig *tls.Config, try
 	return try.Start(newWebsocketDialer(cfg, opts))
 }
 
-// newWebsocketDialer returns a function that
-// can be passed to utils/parallel.Try.Start.
-var newWebsocketDialer = createWebsocketDialer
+// newWebsocketDialer is defined as a variable so that it can be overridden
+// for tests.
+var newWebsocketDialer = newWebsocketDialer0
```

Comment:
I don't think that's really necessary for this change.  If this were all new code, it would be worth talking about, but this is really just a small change to existing code. Reworking how it does all this stuff is not a good use of our time, IMO.